src/index/base.cpp

```diff
@@ -66,7 +66,16 @@ bool BaseIndex::Init()
     } else {
         m_best_block_index = m_chainstate->m_blockman.FindForkInGlobalIndex(active_chain, locator);
     }
+
+    // Note: this will latch to true immediately if the user starts up with an empty
+    // datadir and an index enabled. If this is the case, indexation will happen solely
+    // via `BlockConnected` signals until, possibly, the next restart.
```

Comment:
What do you mean with "until the next restart"? Once the index is in sync with the active chain tip, it will stay that way, regardless of restarts. The only way it could get out of sync is when you disable the index on a restart, advance the tip, then enable it again. Or am I missing something?

Reply:
This is contingent on how fast the particular indexing process is; you can imagine that if a given index takes (for the sake of argument) a minute per block to build, if we get 1000 blocks connected and then the user shuts down immediately, the validationinterface queue may not clear* and we will start up out of sync after having been "synced" in the degenerate way. Does this make sense?

*I know we wait for this queue to clear during a graceful shutdown, but we can't assume this.

Reply:
In commit "index: doc/log BaseIndex sync behavior with empty datadir" (2b48b03f9b6fd3126f957f328dd559fadceeaa7f)

I also found this confusing and would suggest just saying "the index will get updated and stay in sync through BlockConnected notifications"

Reply:
My issue with your suggested comment is that that isn't a complete picture of what's going on - as described above, indexation will only happen via BlockConnectected notifications until the next restart. If we have a hard shutoff that doesn't clear the validation queue, the index will sync sequentially in `ThreadSync()`, which makes that comment somewhat misleading.